packages/core-data/src/resolvers.js

```diff
@@ -165,8 +165,44 @@ export const getEntityRecord =
 					}
 				}
 
-				const record = await apiFetch( { path } );
-				dispatch.receiveEntityRecords( kind, name, record, query );
+				const response = await apiFetch( { path, parse: false } );
+				const record = await response.json();
+
+				const allowHeader = response.headers?.get( 'allow' );
+				const allowedMethods = allowHeader?.allow || allowHeader || '';
+				const permissions = {};
+				const methods = {
+					create: 'POST',
+					read: 'GET',
+					update: 'PUT',
+					delete: 'DELETE',
+				};
+				for ( const [ actionName, methodName ] of Object.entries(
+					methods
+				) ) {
+					permissions[ actionName ] =
+						allowedMethods.includes( methodName );
+				}
```

Comment:
This is getting a bit repetitive. I'm sure we're using the same logic in `canUser()`. Can we deduplicate?

Reply:
That's the plan. I will probably extract those into new utility methods.

Work in progress names:
* `getCapabilitiesFromReponseHeaders` - handles this particular case.
* `getNormalizedCapabilityKey` - handles key generation.

Reply:
Looks good, thanks! As long as it's not a public API, we can be a bit more flexible and not super critical about it.

Reply:
Yes. These are private utilities.